ee/clickhouse/views/person.py

```diff
@@ -1,30 +1,93 @@
-from typing import Callable, Dict, Optional, Tuple
+import json
+from typing import Any, Callable, Dict, List, Optional, Tuple, cast
 
+from django.db.models.query import Prefetch, QuerySet
 from rest_framework.decorators import action
 from rest_framework.exceptions import NotFound
 from rest_framework.request import Request
 from rest_framework.response import Response
 
 from ee.clickhouse.client import sync_execute
 from ee.clickhouse.models.person import delete_person
+from ee.clickhouse.models.property import parse_prop_clauses
 from ee.clickhouse.queries.clickhouse_retention import ClickhouseRetention
 from ee.clickhouse.queries.clickhouse_stickiness import ClickhouseStickiness
 from ee.clickhouse.queries.funnels import ClickhouseFunnelPersons, ClickhouseFunnelTrendsPersons
 from ee.clickhouse.queries.trends.lifecycle import ClickhouseLifecycle
 from ee.clickhouse.sql.person import GET_PERSON_PROPERTIES_COUNT
 from posthog.api.person import PersonViewSet
-from posthog.api.utils import format_next_url, format_offset_absolute_url
-from posthog.constants import INSIGHT_FUNNELS, FunnelVizType
+from posthog.api.utils import format_offset_absolute_url
+from posthog.constants import FunnelVizType
 from posthog.decorators import cached_function
+from posthog.exceptions import UnsupportedFeature
 from posthog.models import Event, Filter, Person
 
+BASE_FILTER_PERSONS_QUERY = """
```

Comment:
This whole query is wrong:

1. It doesn't account for deleted persons or deleted distinct ids
2. It doesn't do data rollups as our other queries do, needed for https://github.com/PostHog/product-internal/issues/134

Reply:
1. :+1: Will add a non-deleted condition, thanks.
2. What's a data rollup in this context?

Reply:
Good question. So you should use ee/clickhouse/sql/person.py#GET_LATEST_PERSON_SQL as a basis rather than rolling your own here. :)

Reply:
Thanks! Didn't have context on this, now reworked the query (I ended up adding `GET_LATEST_PERSON_WITH_DISTINCT_IDS_SQL` to `person.py` `sql/` file, to facilitate search on distinct IDs).